core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/CoarseMesosSchedulerBackend.scala

```diff
@@ -261,7 +269,27 @@ private[spark] class CoarseMesosSchedulerBackend(
     val taskId = status.getTaskId.getValue.toInt
     val state = status.getState
     logInfo(s"Mesos task $taskId is now $state")
+    val slaveId: String = status.getSlaveId.getValue
     stateLock.synchronized {
+      if (TaskState.fromMesos(state).equals(TaskState.RUNNING) &&
```

Comment:
this block of code needs a huge comment:
```
// If the shuffle service is enabled, have the driver register with each one
// of them. This is important for cleaning up state on the shuffle services
// after this application exits; there is currently not a great way to detect
// this through Mesos, since the shuffle services are set up independently.
```